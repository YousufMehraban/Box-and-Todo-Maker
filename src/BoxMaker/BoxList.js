import React, {useState} from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import {v4 as uuid} from 'uuid'

const BoxList = () =>{
    const initialBoxesData = []
    const [Boxes, setBoxes] = useState(initialBoxesData)

    const addBox = (color, width, height)=>{
        setBoxes(boxes => {
            return[...boxes, {color, width, height}]
        })
    }

    const removeBox = (index) => {
        setBoxes(boxes => {
            return boxes.filter((item, idx ) => idx != index)
        })
    }

    return(
        <div>
            <NewBoxForm addBox={addBox}/>
            {Boxes.map((item, idx) => 
                <Box 
                    color={item.color} 
                    width={item.width} 
                    height={item.height}
                    key={uuid()}
                    removeBox={removeBox}
                    idx={idx}
                    />    
            )}
        </div>
    )
}
export default BoxList;


