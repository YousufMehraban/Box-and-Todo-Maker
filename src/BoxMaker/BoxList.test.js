import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it('component runs without breaking', ()=>{
    render(<BoxList />)
})


it('box snapshots should match', ()=>{
    const {asFragment} = render(<BoxList />)
    expect(asFragment()).toMatchSnapshot()
})

it('item should be in the DOM', ()=>{
    const { queryByLabelText } = render(<BoxList />)
    const form = document.querySelector('form')
    const colorInput = queryByLabelText('Color:')
    expect(colorInput).toBeInTheDocument()
    expect(form).toBeInTheDocument()
})

it('items to be added in the DOM', ()=>{
    const {queryByText, queryByLabelText } = render(<BoxList />)
    const colorInput = queryByLabelText('Color:')
    const widthInput = queryByLabelText('Width:')
    const heightInput = queryByLabelText('Height:')
    const submitBTN = queryByText('Submit')
    fireEvent.change(colorInput, {target: { value: 'red'}})
    fireEvent.change(widthInput, {target: { value: '200px'}})
    fireEvent.change(heightInput, {target: { value: '200px'}})
    fireEvent.click(submitBTN)
    const ColorBox = queryByText(0)
    expect(ColorBox).toBeInTheDocument()

    const removeBTN = queryByText('X')
    fireEvent.click(removeBTN)
    expect(ColorBox).not.toBeInTheDocument()
})