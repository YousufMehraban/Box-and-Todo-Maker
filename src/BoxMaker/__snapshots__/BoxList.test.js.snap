// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`box snapshots should match 1`] = `
<DocumentFragment>
  <div>
    <fieldset>
      <legend>
         Add Item: 
      </legend>
      <form
        style="margin: 15px; padding: 15px; border: solid black; background: lightgray; font-weight: bold;"
      >
        <label
          for="Color"
        >
          Color: 
        </label>
        <input
          id="Color"
          name="color"
          placeholder="Color"
          type="text"
          value=""
        />
        <label
          for="width"
        >
          Width: 
        </label>
        <input
          id="width"
          name="width"
          placeholder="width"
          type="text"
          value=""
        />
        <label
          for="height"
        >
          Height: 
        </label>
        <input
          id="height"
          name="height"
          placeholder="height"
          type="text"
          value=""
        />
        <button>
          Submit
        </button>
      </form>
    </fieldset>
  </div>
</DocumentFragment>
`;
